mi/dataset/parser/test/test_ctdbp_cdef_dcl.py

```diff
@@ -382,7 +391,235 @@ def test_bug_11367(self):
+        """
+        with open(os.path.join(RESOURCE_PATH, '20141017.ctdbp3_3rec_w_1hash.log'), 'r') as file_handle:
+
+            num_particles_to_request = 10
+            num_expected_particles = 3
+
+            parser = CtdbpCdefDclParser(True,
+                                        file_handle,
+                                        self.exception_callback)
+
+            particles = parser.get_records(num_particles_to_request)
+
+            # Make sure we obtained expected particle(s)
+            self.assertEquals(len(particles), num_expected_particles)
+            self.assert_particles(particles, "20141017.ctdbp3_3rec_w_1hash.yml", RESOURCE_PATH) 
+
+
+        """
+        test 3 recs with negative pressure
+        2014/10/02 00:30:28.063 [ctdbp3:DLOGP6]: 20.9286,  0.00003,   -0.011, 4130, 1244, 4130, 02 Oct 2014 00:30:23
+        """
+        with open(os.path.join(RESOURCE_PATH, '20141002.ctdbp3_3Rec_negPressure.log'), 'r') as file_handle:
 
+            num_particles_to_request = 10
+            num_expected_particles = 3
+
+            parser = CtdbpCdefDclParser(True,
+                                        file_handle,
+                                        self.exception_callback)
+
+            particles = parser.get_records(num_particles_to_request)
+
+            # Make sure we obtained expected particle(s)
+            self.assertEquals(len(particles), num_expected_particles)
+            self.assert_particles(particles, "20141002.ctdbp3_3Rec_negPressure.yml", RESOURCE_PATH) 
+
+
+        """
+        test 18 recs with one damaged data line
+        """
+        with open(os.path.join(RESOURCE_PATH, '20161025.ctdbp3_damagedRec.log'), 'r') as file_handle:
+
+            num_particles_to_request = 20
+            num_expected_particles = 18
+
+            parser = CtdbpCdefDclParser(True,
+                                        file_handle,
+                                        self.exception_callback)
+
+            particles = parser.get_records(num_particles_to_request)
+
+            # Make sure we obtained expected particle(s)
+            self.assertEquals(len(particles), num_expected_particles)
+            self.assert_particles(particles, "20161025.ctdbp3_damagedRec.yml", RESOURCE_PATH) 
+
+
+
+        """
+        test large file, 24 recs 
+        """
+        with open(os.path.join(RESOURCE_PATH, '20140928.ctdbp3_24rec.log'), 'r') as file_handle:
+
+            num_particles_to_request = 30
+            num_expected_particles = 24
+
+            parser = CtdbpCdefDclParser(True,
+                                        file_handle,
+                                        self.exception_callback)
+
+            particles = parser.get_records(num_particles_to_request)
+
+            # Make sure we obtained expected particle(s)
+            self.assertEquals(len(particles), num_expected_particles)
+            self.assert_particles(particles, "20140928.ctdbp3_24rec.yml", RESOURCE_PATH) 
+
```

Comment:
please use an editor that enforces PEP8 standards (in this case too many blank lines in a function body)